lib/client/db/mysql/repl/commands.go

```diff
@@ -0,0 +1,284 @@
+	shortcutPrefix string
+}
+
+// command represents a command that can be executed in the REPL.
+type command struct {
+	// name is the command's full name.
+	name string
+	// shortcut is the command's escaped single char name: \x.
+	shortcut rune
+	// description provides a user-friendly explanation of what the command
+	// does.
+	description string
+	// execFunc is the function to execute the command. The commands can either
+	// return a reply (that will be sent back to the client) as a string. It can
+	// terminate the REPL by returning bool on the second argument.
+	execFunc func(r *REPL, args string) (reply string, exit bool)
+}
+
+func (c *command) checkAndSetDefaults() error {
+	if len(c.name) == 0 {
+		return trace.BadParameter("missing command name")
+	}
+	if c.shortcut == 0 {
+		return trace.BadParameter("missing command short name")
+	}
+	if len(c.description) == 0 {
+		return trace.BadParameter("missing command description")
+	}
+	if c.execFunc == nil {
+		return trace.BadParameter("missing command exec func")
+	}
+	c.name = normalizeCommandName(c.name)
+	return nil
+}
+
+func normalizeCommandName(name string) string {
+	return strings.ToLower(name)
+}
+
+func newCommands() (*commandManager, error) {
+	c := &commandManager{
+		byName:         make(map[string]*command),
+		byShortcut:     make(map[rune]*command),
+		shortcutPrefix: `\`,
+	}
+	for _, cmd := range []*command{
+		{
+			name:        "delimiter",
+			shortcut:    'd',
+			description: "Set statement delimiter. With no argument the delimiter resets to the default.",
+			execFunc: func(r *REPL, args string) (reply string, exit bool) {
+				delim := getArg(args)
+				if err := r.lex.setDelimiter(delim); err != nil {
+					return err.Error(), false
+				}
+				return "", false
+			},
+		},
+		{
+			name:        "help",
+			shortcut:    'h',
+			description: "Show the list of supported commands.",
+			execFunc: func(r *REPL, _ string) (string, bool) {
+				return r.commands.help(), false
+			},
+		},
+		{
+			name:        "quit",
+			shortcut:    'q',
+			description: "Terminates the session.",
+			execFunc:    func(_ *REPL, _ string) (string, bool) { return "", true },
+		},
+		{
+			name:        "session",
+			shortcut:    's',
+			description: "Display information about the current session, like user, and database instance.",
+			execFunc: func(r *REPL, _ string) (string, bool) {
+				return fmt.Sprintf("Connected to %q instance using database %q as user %q.",
+					r.route.ServiceName,
+					formatDatabaseName(r.route.Database),
+					r.route.Username,
+				), false
+			},
+		},
+		{
+			name:        "teleport",
+			shortcut:    't',
+			description: "Show Teleport interactive shell information, such as execution limitations.",
+			execFunc: func(_ *REPL, _ string) (string, bool) {
+				var sb strings.Builder
+				for _, l := range descriptiveLimitations {
+					sb.WriteString("- " + strings.Join(strings.Split(l, "\n"), "\n  ") + lineBreak)
```

Comment:
Could this split and join operation be simplified with a call to strings.Replace?